Fix not specified array as primary option for "function-url-scheme-whitelist"

hudochenkov opened the pull request:
<!---
Please read the following. Pull requests that do not adhere to these guidelines will be closed.

Each pull request must, with the exception of minor documentation fixes, be associated with an open issue. If a corresponding issue does not exist please stop. Instead, create an issue so we can discuss the change first.

If there is an associated open issue, then the next step is to make sure you've read the relevant developer guide:

- Adding a rule: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#write-the-rule

- Adding an option: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules

- Fixing a bug: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs

Once you've done that, then please continue by answering these two questions:  -->

> Which issue, if any, is this issue related to?

https://github.com/stylelint/stylelint/issues/2446

> Is there anything in the PR that needs further explanation?

For some strange reason it wasn't defined, that rule can accept an array as primary option. Test didn't break, though. I wonder why? I believe we should have tests to test this behavior, but I have no idea how to do them.

I've tested results before and after this change via Node API. It stopped throwing warnings.

evilebottnawi commented:
/cc @davidtheclark

davidtheclark commented:
@hudochenkov That property ends up being used in https://github.com/stylelint/stylelint/blob/01167e6ed006a9cd2fe8e0777e84b3d07736b4d2/lib/normalizeRuleSettings.js#L23. We could run try settings for tests through this function, to ensure that the settings in tests undergo the same processing they do in the real program.

evilebottnawi commented:
@davidtheclark it is be good

hudochenkov commented:
@davidtheclark I'm afraid I didn't understand your idea :(

davidtheclark commented:
@hudochenkov Could you merge this PR and open up an issue for follow-up, and I will try to find some time this weekend to try some things?

hudochenkov commented:
Added to changelog:

-   Fixed: `function-url-scheme-whitelist` was working incorrectly if more than one URL scheme were specified ([#2447](https://github.com/stylelint/stylelint/pull/2447)).
